### PR TITLE
MMCA 5058 - Introduce Excise deferment 1920 and Duty deferment 1720 statements on Duty deferment frontend service

### PR DESCRIPTION
**Pull request by HariHmrc**
- case statement inside view components duty_deferment_accordian_content and duty_deferment_head are NOT refactored to viewModel due to points / time constraints

- spec files for duty_deferment_accordian_content and duty_deferment_head view components  are not yet done due to time constraint. However, its being implemented as part of ticket 'https://jira.tools.tax.service.gov.uk/browse/MMCA-5069' in next sprint (sprint M)

### CODE REVIEW

File: conf/messages.en [hunk 1]
```diff
@@ -8,10 +8,12 @@ cf.account.NiAccount=Account: {0} (Northern Ireland)
 cf.account.detail.title=Duty deferment statements
 cf.account.detail.deferment-account-heading=Duty deferment statements
 cf.account.details.previous-eori=EORI: {0}
-cf.account.detail.direct-debit.duty-vat-and-excise=HMRC will charge your bank each month for Duty and VAT on or after the 16th, and Excise on or before the 29th.
+cf.account.detail.direct-debit.duty-vat-and-excise=HMRC will collect duty and VAT payments mid-month and excise payments near the end of the month.
 cf.account.detail.period-group={0} to {1} {2}
 cf.account.detail.row.supplementary.info=Supplementary end of month
 cf.account.details.row.excise.info=Excise summary
+cf.account.details.row.ExciseDeferment.info=Excise deferment 1920
+cf.account.details.row.DutyDeferment.info=Duty deferment 1920
```
Comment (tyagikumarvineet) [inline conf/messages.en:16]: the value for this msg key  should be **`Duty deferment 1720`**

File: app/views/components/duty_deferment_accordian_content.scala.html [hunk 2]
```diff
@@ -23,26 +23,30 @@
...
+        <dt class="govuk-summary-list__value"
+            id="statements-list-@{historyIndex}-@{group.year}-@{group.month}-row-@{index}-date-cell">
+            @{
+                period.defermentStatementType match {
+                    case Supplementary => messages("cf.account.detail.row.supplementary.info")
+                    case Excise => messages("cf.account.details.row.excise.info")
+                    case ExciseDeferment => messages("cf.account.details.row.ExciseDeferment.info")
+                    case DutyDeferment => messages("cf.account.details.row.DutyDeferment.info")
+                    case _ =>
+                        messages("cf.account.detail.period-group",
```
Comment (tyagikumarvineet) [inline app/views/components/duty_deferment_accordian_content.scala.html:37]: better to format like below () to make it more readable

```
messages("cf.account.detail.period-group",
                    Formatters.dateAsDay(period.startDate),
                    Formatters.dateAsDay(period.endDate),
                    Formatters.dateAsMonth(period.endDate)
                  )
```

File: app/views/duty_deferment_account/duty_deferment_head.scala.html [hunk 3]
```diff
@@ -21,35 +21,39 @@
...
+        <dt id="statements-list-0-@{group.year}-@{group.month}-row-@{index}-date-cell"
+            class="govuk-summary-list__value">
+            @{
+                period.defermentStatementType match {
+                    case Supplementary => messages("cf.account.detail.row.supplementary.info")
+                    case Excise => messages("cf.account.details.row.excise.info")
+                    case ExciseDeferment => messages("cf.account.details.row.ExciseDeferment.info")
+                    case DutyDeferment => messages("cf.account.details.row.DutyDeferment.info")
+                    case _ =>
+                        messages("cf.account.detail.period-group",
```
Comment (tyagikumarvineet) [inline app/views/duty_deferment_account/duty_deferment_head.scala.html:44]: Better to format like below to make it more readable

```
messages("cf.account.detail.period-group",
                   Formatters.dateAsDay(period.startDate),
                    Formatters.dateAsDay(period.endDate),
                    Formatters.dateAsMonth(period.endDate)
)
```

File: test/models/SdesFileSpec.scala
Comment (tyagikumarvineet) [inline test/models/SdesFileSpec.scala:1]: Better to add the tests for unapply method for the new types

**Review by tyagikumarvineet — DISMISSED** [review]

File: app/views/components/duty_deferment_accordian_content.scala.html [hunk 4]
```diff
@@ -23,26 +23,31 @@
...
+        </dt>
+        <dd class="govuk-summary-list__actions">
+            @duty_deferment_file(period, Pdf, s"pdf-${historyIndex}-${group.year}-${group.month}-row-${index}")
+        </dd>
+        <dd class="govuk-summary-list__actions">
+            @duty_deferment_file(period, Csv, s"csv-${historyIndex}-${group.year}-${group.month}-row-${index}")
+        </dd>
+    </div>
+    }
 </dl>
```
Comment (AdamCoops) [inline app/views/components/duty_deferment_accordian_content.scala.html:53]: Can we add a newline after the </dl> to remove the end of file error

**Review by AdamCoops — APPROVED** [review]
approved

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**